File: CS/DXScheduler_PopUpMenuCustomization/ViewModel/MainViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using System;
using System.Collections.ObjectModel;

namespace DXScheduler_PopUpMenuCustomization {
    [POCOViewModel]
    public class MainViewModel {
        readonly SportChannelsData data = new SportChannelsData(2);
        protected MainViewModel() {
            WorkTimeStart = TimeSpan.FromHours(6);
            WorkTimeEnd = TimeSpan.FromHours(21);
        }
        public virtual TimeSpan WorkTimeStart { get; set; }
        public virtual TimeSpan WorkTimeEnd { get; set; }
        public virtual TimeSpanRange WorkTime { get; set; }
        public virtual ObservableCollection<SportEvent> SportEvents { get { return data.Events; } }
        public virtual ObservableCollection<SportChannel> SportChannels { get { return data.Channels; } }
        public virtual ObservableCollection<SportGroup> SportGroups { get { return data.Groups; } }
        public virtual ObservableCollection<SportEvent> CancelledEvents { get; set; }

        protected void OnWorkTimeStartChanged() {
            var end = (WorkTimeEnd < WorkTimeStart) ? WorkTimeStart.Add(TimeSpan.FromHours(1)) : WorkTimeEnd;
            WorkTime = new TimeSpanRange(WorkTimeStart, end);
        }
        protected void OnWorkTimeEndChanged() {
            var start = (WorkTimeEnd < WorkTimeStart) ? WorkTimeEnd.Add(TimeSpan.FromHours(-1)) : WorkTimeStart;
            WorkTime = new TimeSpanRange(start, WorkTimeEnd);
        }
    }
}

File: VB/DXScheduler_PopUpMenuCustomization/ViewModel/MainViewModel.vb
#Region "#viewmodel"
Imports System
Imports System.Collections.ObjectModel
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.Mvvm

Namespace DXScheduler_PopUpMenuCustomization

    <POCOViewModel>
    Public Class MainViewModel

        Private ReadOnly data As SportChannelsData = New SportChannelsData(2)

        Protected Sub New()
            WorkTimeStart = TimeSpan.FromHours(6)
            WorkTimeEnd = TimeSpan.FromHours(21)
        End Sub

        Public Overridable Property WorkTimeStart As TimeSpan

        Public Overridable Property WorkTimeEnd As TimeSpan

        Public Overridable Property WorkTime As TimeSpanRange

        Public Overridable ReadOnly Property SportEvents As ObservableCollection(Of SportEvent)
            Get
                Return data.Events
            End Get
        End Property

        Public Overridable ReadOnly Property SportChannels As ObservableCollection(Of SportChannel)
            Get
                Return data.Channels
            End Get
        End Property

        Public Overridable ReadOnly Property SportGroups As ObservableCollection(Of SportGroup)
            Get
                Return data.Groups
            End Get
        End Property

        Public Overridable Property CancelledEvents As ObservableCollection(Of SportEvent)

        Protected Sub OnWorkTimeStartChanged()
            Dim [end] = If(WorkTimeEnd < WorkTimeStart, WorkTimeStart.Add(TimeSpan.FromHours(1)), WorkTimeEnd)
            WorkTime = New TimeSpanRange(WorkTimeStart, [end])
        End Sub

        Protected Sub OnWorkTimeEndChanged()
            Dim start = If(WorkTimeEnd < WorkTimeStart, WorkTimeEnd.Add(TimeSpan.FromHours(-1)), WorkTimeStart)
            WorkTime = New TimeSpanRange(start, WorkTimeEnd)
        End Sub
    End Class
End Namespace
#End Region  ' #viewmodel
